active/000-buisness-attributes/000-business-attributes.md

```diff
@@ -0,0 +1,258 @@
+2. Ability to update a business attribute record centrally
+3. Ability to track any discrepancies between business attributes and associated fields
+
+#### Good-to-Haves
+1. Approval workflow for creation/updation
+
+### Dataset Owner
+
+#### Must Haves
+1. Ability to inherit all attributes from a business attribute to a field by attaching the business attribute to the field
+1. Ability to inherit all updates from a business attribute to a field
+
+#### Good-to-Haves
+
+1. Ability to intelligently inherit attributes from business attributes to fields in datasets via system auto-assignment
+
+### Business User
+
+#### Must Haves
+1. Ability to search for fields using business description/tags/glossary attached to business attribute
+
+
+## Non-Requirements
+*  Allow dataset owners to propose changes to business attributes
+* Allow dataset owners to partially inherit from business attributes (eg only inherit description but not tags)
+* Allow fields to have multiple business attributes
+* Update descriptions in data sources
+
+## Detailed design
+
+### Sample Mockups
+
+#### Show Business Atrribute Launcher 
+![](business-attribute-rfc-1.png)
+
+#### List Business Atrributes
+![](business-attribute-rfc-2.png)
+
+#### Create Business Attribute  
+![](business-attribute-rfc-3.png)
+
+#### Business Attribute Details
+![](business-attribute-rfc-4.png)
+
+#### Enrich Schema
+Current feature of enriching the schema field of dataset remains as it is
+
+![](business-attribute-rfc-5.png)
+
+#### Attaching Business Attribute
+![](business-attribute-rfc-6.png)
+
+#### Metadata Model Enhancements
+
+Below are the current Relationships of Dataset Entity with its aspects in Datahub.
+
+![](business-attribute-rfc-7.png)
+
+
+### Model Business Attribute Entity
+
+We are proposing to introduce a new entity `BusinessAttribute`, which will use a new Aspect `BusinessAttributeInfo`. This new entity will be a top level entity to allow it to be defined and controlled independently by the business team.
+
+User can able to attach `Business Attributes` only to `Schema Fields`. We will make necessary UI changes to control this feature.
+
+New `BusinessAttributeInfo` aspect contains existing `EditableSchemaFieldInfo` record type and also `customProperties`.  Also we are introducing one new record type, `PhysicalEditableSchemaFieldInfo` which includes existing `EditableSchemaFieldInfo`. As per this implementation, existing aspect `EditableSchemaMetadata` now contains a list of `PhysicalEditableSchemaFieldInfo` records instead of `EditableSchemaFieldInfo` records.
+
+Each `PhysicalEditableSchemaFieldInfo` will contain an `EditableSchemaFieldInfo` and the new field "`businessAttribute`" to represent the link between dataset schema field and business attribute. We are also proposing the introduction of a new field, "`type`", in `EditableSchemaFieldInfo`.
```

Comment:
What if we just had a new aspect (similar to globalTags and glossaryTerms) called `businessAttributes` attached to the `schemaField` entity and that aspect contained a list of business attribute urns that were associated with the schema field entity? Would that simplify the representation?

Reply:
A schema-field is represents a unique Semantic field or Business Attribute. We have taken your feedback in other comment and simplified the Record structures.